packages/ui-components/src/components/windowedlist.ts

```diff
@@ -1439,16 +1620,52 @@ export namespace WindowedList {
    * Windowed list view constructor options
    */
   export interface IOptions<
-    T extends WindowedList.IModel = WindowedList.IModel
+    T = any,
+    U extends WindowedList.IModel<T> = WindowedList.IModel<T>
   > {
     /**
      * Windowed list model to display
      */
-    model: T;
+    model: U;
     /**
      * Windowed list layout
      */
     layout?: WindowedLayout;
+
+    /**
+     * A renderer for the elements of the windowed list.
+     */
+    renderer?: IRenderer<T>;
+
+    /**
+     * Whether the windowed list should display a scrollbar UI.
+     */
+    scrollbar?: boolean;
+  }
+
+  /**
+   * A windowed list element renderer.
+   */
+  export interface IRenderer<T = any> {
+    /**
+     * Create the outer, root element of the windowed list.
+     */
+    createOuter(): HTMLElement;
+
+    /**
+     * Create the virtual scrollbar element.
+     */
+    createScrollbar(): HTMLUListElement;
```

Comment:
I would argue it should be an ordered list:

```suggestion
    createScrollbar(): HTMLOListElement;
```